pkg/grafana/types.go

```diff
@@ -1,6 +1,46 @@
 package grafana
 
 type Organization struct {
-	ID   int64
-	Name string
+	ID       int64
+	Name     string
+	TenantID string
+}
+
+type Datasource struct {
+	ID        int64
+	Name      string
+	IsDefault bool
+	Type      string
+	URL       string
+	Access    string
+	JSONData  map[string]interface{}
+}
+
+func (d Datasource) withID(id int64) Datasource {
+	return Datasource{
+		ID:        id,
+		Name:      d.Name,
+		IsDefault: d.IsDefault,
+		Type:      d.Type,
+		URL:       d.URL,
+		Access:    d.Access,
+		JSONData:  d.buildJSONData(),
+	}
```

Comment:
No need to copy data on non-pointer receivers
```suggestion
	datasource.ID = id
	return d
```

Reply:
That does not update the current state? This is a method so I was not sure